File: BookingSystem/DTOs/Vendor.cs
using System;
using System.Collections.Generic;
namespace BookingSystem.DTOs;

public class VendorDashboardDto
{
    public int TotalBookings { get; set; }
    public int PendingBookings { get; set; }
    public decimal TotalRevenue { get; set; }
    public decimal AverageRating { get; set; }

    public List<TopServiceDto> TopServices { get; set; } = new();
    public List<RecentReviewDto> RecentReviews { get; set; } = new();
}

public class TopServiceDto
{
    public int Rank { get; set; }
    public string ServiceName { get; set; } = string.Empty;
    public decimal Price { get; set; }
    public int BookingCount { get; set; }
}

public class RecentReviewDto
{
    public decimal Rating { get; set; }
    public string? Comment { get; set; }
    public string CustomerName { get; set; } = string.Empty;
    public string? PetName { get; set; }
    public DateTime CreatedAt { get; set; }
}

File: booking-ui/src/type/vendor.ts
// src/types/vendor.ts
export interface TopService {
    rank: number;
    serviceName: string;
    price: number;
    bookingCount: number;
}

export interface RecentReview {
    rating: number;
    comment?: string;
    customerName: string;
    petName?: string;
    createdAt: string;
}

export interface VendorDashboardData {
    totalBookings: number;
    pendingBookings: number;
    totalRevenue: number;
    averageRating: number;
    topServices: TopService[];
    recentReviews: RecentReview[];
}